src/components/HomePage/ListOfHouses/ListOfHouses.tsx

```diff
@@ -68,26 +77,50 @@ function ListOfHouses({ houses }: { houses: BasicHouseData[] }) {
         />
       </Box>
       <Box component="div" className={styles.housesList}>
-        {sortedHouses.map((item: BasicHouseData, i) => (
-          <Box component="div" className={styles.houseElement} key={i.toString()}>
-            <h4>
-              {item.city}, {item.streetName} {item.streetNumber}
-            </h4>
-            <p className={styles.price}>
-              {item.price}zł/m<sup>2</sup>
-            </p>
-            <img src={item.photo_0 ? item.photo_0 : noPhoto} alt="House" />
-            <p className={styles.shortInfo}>{item.descriptionField}</p>
-            <Button className={styles.moreInfo}>more info</Button>
-            <Checkbox
-              color="warning"
-              {...label}
-              icon={<FavoriteBorderIcon />}
-              checkedIcon={<FavoriteIcon />}
-              className={styles.icon}
-            />
-          </Box>
-        ))}
+        {sortedHouses.map((item: BasicHouseData, i) => {
+          if (item.images.length === 0) {
+            item.images.push(noPhoto);
+          }
+          if (item.images.length === 1) {
+            slideProperties.arrows = false;
+            slideProperties.canSwipe = false;
+          }
+          return (
+            <Box component="div" className={styles.houseElement} key={i.toString()}>
+              <h4>
+                {item.city}, {item.street} {item.houseNr}
+              </h4>
+              <p className={styles.price}>
+                {item.price}zł/m<sup>2</sup>
+              </p>
+              <Slide {...slideProperties}>
+                {item.images.map((image, index) => {
+                  if (image) {
+                    return (
+                      <div className="each-slide" key={index.toString()}>
+                        <img src={image} alt="House" />
+                      </div>
+                    );
+                  }
+                  return (
```

Comment:
I wonder if this situation is possible as you already add `noPhoto` to the array of images if it's empty.